packages/orchestration/src/examples/send-anywhere.flows.js

```diff
@@ -112,20 +125,36 @@ export const sendIt = async (
       );
       void log(`completed transfer to ${destAddr}`);
     } catch (e) {
-      await withdrawToSeat(sharedLocalAccount, seat, give);
-      const errorMsg = `IBC Transfer failed ${q(e)}`;
-      void log(`ERROR: ${errorMsg}`);
-      seat.fail(errorMsg);
-      throw makeError(errorMsg);
+      return recoverFailedTransfer(e);
     }
   } else if (amt.brand === USDC && 'cctpDestinationDomain' in info) {
     const nobleAccount = await nobleAccountP;
     const nobleAddr = nobleAccount.getAddress();
     const denomAmt = { denom, value: amt.value };
-    await sharedLocalAccount.transfer(nobleAddr, denomAmt);
+    try {
+      await sharedLocalAccount.transfer(nobleAddr, denomAmt);
+    } catch (e) {
+      return recoverFailedTransfer(e);
+    }
     void log('assets are now on noble');
-
-    Fail`not yet unimplemented`;
+    const destAccountId = /** @type {const} */ (
+      `${info.namespace}:${info.reference}:${destAddr}`
+    );
+    try {
+      await nobleAccount.depositForBurn(destAccountId, denomAmt);
+    } catch (e) {
+      try {
+        // transfer back to agoric contract account if depositForBurn is not accepted by noble chain
+        const lcaAddress = await sharedLocalAccount.getAddress();
+        await nobleAccount.transfer(lcaAddress, denomAmt);
+      } catch (err) {
+        const errorMsg = `IBC Transfer from Noble back to LocalAccount failed: ${q(err)}`;
```

Comment:
Customer funds are stuck in the `nobleAccount` in this case, right? Seems worth calling out. Something about "contract upgrade required to recover funds".